### PR TITLE
Migrate wingdi and winspool APIs

### PR DESCRIPTION
**Pull request by skdltmxn**

### CODE REVIEW

File: src/um/wingdi.rs [hunk 1]
```diff
@@ -2010,23 +2045,41 @@ ENUM!{enum DISPLAYCONFIG_COLOR_ENCODING {
...
+]}
+UNION2!{union DISPLAYCONFIG_GET_ADVANCED_COLOR_INFO_u {
+    [u32; 1],
+    s s_mut: DISPLAYCONFIG_GET_ADVANCED_COLOR_INFO_u_s,
+    value value_mut : UINT32,
+}}
 STRUCT!{struct DISPLAYCONFIG_GET_ADVANCED_COLOR_INFO {
     header: DISPLAYCONFIG_DEVICE_INFO_HEADER,
-    value: UINT32,
+    u: DISPLAYCONFIG_GET_ADVANCED_COLOR_INFO_u,
```
Comment (retep998) [inline src/um/wingdi.rs]: In this case, where it is a union with only two variants, where one is a bitfield, and the other is the same type as the bitfield, I prefer to skip having a union and simply have the bitfield applied directly to that field. The layout is the same, but it makes working with the type easier.

File: src/um/wingdi.rs [hunk 2]
```diff
@@ -2633,9 +2687,606 @@ extern "system" {
...
-*********/
+FN!{stdcall LPFNDEVMODE(
+    HWND,
+        HMODULE,
+        LPDEVMODEA,
+        LPSTR,
+        LPSTR,
+        LPDEVMODEA,
+        LPSTR,
+        UINT,
```
Comment (retep998) [inline src/um/wingdi.rs]: What happened to the indentation here?

File: src/um/winspool.rs [hunk 3]
```diff
@@ -36,3 +2146,283 @@ STRUCT!{struct PRINTER_OPTIONSW {
...
+ENUM!{enum EPrintXPSJobOperation {
+    kJobProduction = 1,
+    kJobConsumption,
+}}
+STRUCT!{struct PrintPropertyValue_value_propertyBlob {
+    cbBuf: DWORD,
+    pBuf: LPVOID,
+}}
+UNION2!{union PrintPropertyValue_value {
+    [usize; 2],
```
Comment (retep998) [inline src/um/winspool.rs]: For 32-bit this is actually `[u64; 1]`.

File: src/um/winspool.rs [hunk 4]
```diff
@@ -36,3 +2146,283 @@ STRUCT!{struct PRINTER_OPTIONSW {
...
+    pszDriverName: LPWSTR,
+}}
+pub type PPRINTER_CONNECTION_INFO_1W = *mut PRINTER_CONNECTION_INFO_1W;
+pub type LPPRINTER_CONNECTION_INFO_1W = *mut PRINTER_CONNECTION_INFO_1W;
+extern "system" {
+    pub fn AddPrinterConnection2A(
+        hWnd: HWND, pszName: LPCSTR, dwLevel: DWORD, pConnectionInfo: PVOID,
+    ) -> BOOL;
+    pub fn AddPrinterConnection2W(
+        hWnd: HWND, pszName: LPCWSTR, dwLevel: DWORD, pConnectionInfo: PVOID,
```
Comment (retep998) [inline src/um/winspool.rs]: Block indentation!

File: src/um/winspool.rs (hunk 3 above)
Comment (skdltmxn) [inline src/um/winspool.rs]: Is it different between ```[usize; 2]``` and ```[u64; 1]```?
I think both are 8 bytes on 32-bit.

Comment (retep998) [inline src/um/winspool.rs]: For 32-bit platforms `usize` has an alignment of 4 while `u64` has an alignment of 8. It's an important difference.

**Comment by skdltmxn** [comment]
Fixed!

File: src/um/winspool.rs [hunk 5]
```diff
@@ -36,3 +2146,289 @@ STRUCT!{struct PRINTER_OPTIONSW {
...
+    kJobProduction = 1,
+    kJobConsumption,
+}}
+STRUCT!{struct PrintPropertyValue_value_propertyBlob {
+    cbBuf: DWORD,
+    pBuf: LPVOID,
+}}
+UNION2!{union PrintPropertyValue_value {
+    [u64; 1] [u64; 2],
+    propertyByte propertyByte_mut :BYTE,
```
Comment (retep998) [inline src/um/winspool.rs]: Spacing wrong.

Comment (skdltmxn) [inline src/um/winspool.rs]: Oops, fixed it.

**Pull request merged**